hil-test/src/bin/esp_radio_init.rs

```diff
@@ -249,4 +292,109 @@ mod tests {
         info!("Low: wait for tasks to finish");
         test_context.ready_semaphore.take(None);
     }
+
+    #[test]
+    #[cfg(multi_core)]
+    fn test_esp_preempt_smp(p: Peripherals) {
+        use core::ffi::c_void;
+
+        use esp_hal::system::Cpu;
+        use esp_radio_preempt_driver as preempt;
+        use preempt::semaphore::{SemaphoreHandle, SemaphoreKind};
+
+        let sw_ints = SoftwareInterruptControl::new(p.SW_INTERRUPT);
```

Comment:
I'm wondering if we should treat the FROM_CPUn as separate peripheral singletons.